### PR TITLE
Changes to enable LDAP synchronization

### PR DESCRIPTION
**Pull request by amithkanand**
This PR addes required changes to enable `Ambari-Ldap` integration.

### CODE REVIEW

**Comment by amithkanand** [comment]
The PR has been tested successfully on a physical cluster.

File: cookbooks/ambari/attributes/default.rb [hunk 1]
```diff
@@ -20,82 +20,106 @@
...
-node.default['ambari']['webhcat.port'] = '50111'
-node.default['ambari']['webhcat.hostname'] = 'u1203.ambari.apache.org'
-node.default['ambari']['webhcat.port'] = '50111'
+default['ambari']['webhcat.hostname'] = 'u1203.ambari.apache.org'
+default['ambari']['webhcat.port'] = '50111'
+default['ambari']['webhcat.hostname'] = 'u1203.ambari.apache.org'
+default['ambari']['webhcat.port'] = '50111'
+
+# Ambari LDAP Sync attributes
+default['ambari']['ldap']['groups_to_sync']=''
```
Comment (cbaenziger) [inline cookbooks/ambari/attributes/default.rb]: Here and below, some rubocop-isms...

File: cookbooks/ambari/recipes/ambari_ldap_sync.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+if node['ambari']['ambari.ldap.isConfigured'] then
+  pwd_file=node['ambari']['ldap']['password_file']
```
Comment (cbaenziger) [inline cookbooks/ambari/recipes/ambari_ldap_sync.rb]: More inconsistencies with `foo = bar` and `foo=bar`...

File: cookbooks/ambari/recipes/ambari_ldap_sync.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  grp_file=node['ambari']['ldap']['groups_file']
+  cfg_dir=node['ambari']['ambari_server_conf_dir']
+  groupsFile=File.join(cfg_dir, grp_file )
+  passwordFile=File.join(cfg_dir, pwd_file)
+  ldap_pwd = node['ambari']['ldap_password'] 
+
+  file passwordFile do
+    content "#{ldap_pwd}"
```
Comment (cbaenziger) [inline cookbooks/ambari/recipes/ambari_ldap_sync.rb]: Any reason you need string interpolation here?

File: cookbooks/ambari/recipes/ambari_ldap_sync.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    mode 0444
+    sensitive true
+  end
+
+  file groupsFile do
+    content node['ambari']['ldap']['groups_to_sync']
+    mode 044
+    sensitive true
+    notifies :run, "execute[SyncLdapData]", :delayed
```
Comment (cbaenziger) [inline cookbooks/ambari/recipes/ambari_ldap_sync.rb]: Unnecessary `"` here, please rubocop-ify

File: cookbooks/bach_ambari/attributes/default.rb [hunk 5]
```diff
@@ -16,18 +16,19 @@
...
 mysql_port = node['bcpc']['hadoop']['mysql_port'] || 3306
 
-node.force_default['ambari']['databaseport'] = "#{mysql_port}"
-# node.force_default['ambari']['databasehost'] = mysql_hosts
-node.force_default['ambari']['databasename'] = 'ambari'
-node.force_default['ambari']['databaseusername'] = 'ambari'
-node.force_default['ambari']['databasepassword'] = get_config('mysql-ambari-password') || get_config('password', 'mysql-ambari', 'os')
-node.default['ambari']['mysql_schema_path'] = '/var/lib/ambari-server/resources/Ambari-DDL-MySQL-CREATE.sql'
+force_default['ambari']['databaseport'] = "#{mysql_port}"
+# force_default['ambari']['databasehost'] = mysql_hosts
```
Comment (cbaenziger) [inline cookbooks/bach_ambari/attributes/default.rb]: Any reason for this comment here?

File: cookbooks/ambari/recipes/ambari_ldap_sync.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
+if node['ambari']['ambari.ldap.isConfigured'] then
+  pwd_file=node['ambari']['ldap']['password_file']
+  grp_file=node['ambari']['ldap']['groups_file']
+  cfg_dir=node['ambari']['ambari_server_conf_dir']
+  groupsFile=File.join(cfg_dir, grp_file )
+  passwordFile=File.join(cfg_dir, pwd_file)
+  ldap_pwd = node['ambari']['ldap_password']
```
Comment (cbaenziger) [inline cookbooks/ambari/recipes/ambari_ldap_sync.rb]: I believe this is a nil look-up. Am I missing the default value for `node['ambari']['ldap_password']` some where in the Ambari cookbook?

File: cookbooks/ambari/recipes/ambari_server_setup.rb [hunk 7]
```diff
@@ -20,7 +20,8 @@
 
 
 # /etc/ambari-server/conf/ambari.properties
-template File.join(node['ambari']['ambari_server_conf_dir'],'ambari.properties') do
+template "Creating ambari properties file" do
```
Comment (cbaenziger) [inline cookbooks/ambari/recipes/ambari_server_setup.rb]: More rubocop violations...

File: cookbooks/bach_ambari/attributes/default.rb [hunk 8]
```diff
@@ -16,18 +16,19 @@
...
 
 mysql_port = node['bcpc']['hadoop']['mysql_port'] || 3306
 
-node.force_default['ambari']['databaseport'] = "#{mysql_port}"
-# node.force_default['ambari']['databasehost'] = mysql_hosts
-node.force_default['ambari']['databasename'] = 'ambari'
-node.force_default['ambari']['databaseusername'] = 'ambari'
-node.force_default['ambari']['databasepassword'] = get_config('mysql-ambari-password') || get_config('password', 'mysql-ambari', 'os')
-node.default['ambari']['mysql_schema_path'] = '/var/lib/ambari-server/resources/Ambari-DDL-MySQL-CREATE.sql'
+force_default['ambari']['databaseport'] = "#{mysql_port}"
```
Comment (cbaenziger) [inline cookbooks/bach_ambari/attributes/default.rb]: Do you need a string interpolation here?

**Comment by cbaenziger** [comment]
Talking with @dbist I understand we may need to permission users for Views too; will this be necessary or have you found users are able to access WFM and the like as soon as they login with this?

File: cookbooks/bach_ambari/recipes/default.rb [hunk 9]
```diff
@@ -18,47 +18,66 @@
 #
 
 ambari_admin_password = make_config('ambari-admin-password', secure_password)
-node.force_default['ambari']['admin']['password'] = ambari_admin_password
+force_default['ambari']['admin']['password'] = ambari_admin_password
```
Comment (aespinosa) [inline cookbooks/bach_ambari/recipes/default.rb]: I think this should still be `node.force_default...` since this is recipe

**Pull request merged**